pcdsdevices/interface.py

```diff
@@ -1646,6 +1656,77 @@ def transmission(self):
                 return 1
 
 
+class LightpathMixin(Device):
+    """
+    Mix-in class that makes it easier to establish a lightpath interface.
+
+    Gathers relevant signals into a SummarySignal that can be
+    subscribed to for monitoring the lightpath-relevant state.
+
+    Users must implement ``get_lightpath_status``, and return a
+    LightpathStatus object
+    Create an object similar to the following:
+
+    class MyDevice(LightpathMixin):
+        lp_signals = ['sig1', 'sig2']
+
+        def get_lightpath_status(self):
+            # Logic, calculations for transmission, etc
+            status = LightpathStatus(
+                inserted=True, removed=False, transmission=0.0,
+                output_branch='L3'
+            )
+            return status
+    """
+    # Component names whose values are relevant for inserted/removed
+    lightpath_cpts = []
+
+    # Flag to signify that subclass is another mixin, rather than a device
+    _lightpath_mixin = False
+
+    # Mixin holds one summary signal that changes with lightpath_cpts
+    lp_summary = Cpt(SummarySignal, name='lp_summary')
+
+    def __init__(self, *args,
+                 input_branches=[], output_branches=[], **kwargs):
+        self._lightpath_values = {}
+        self._lightpath_ready = False
+        self._retry_lightpath = False
+        self.input_branches = input_branches
+        self.output_branches = output_branches
+        # TODO: grab destination / source lists, if they exist?
+        super().__init__(*args, **kwargs)
+        for sig in self.lightpath_cpts:
+            self.lp_summary.add_signal_by_attr_name(sig)
+
+        if not self.input_branches or not self.output_branches:
+            raise NotImplementedError(
+                'Did not implement LightpathMixin properly.  '
+                'Must supply input and output branches.'
+            )
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        if cls._lightpath_mixin:
+            # Child of cls will inherit this as False
+            cls._lightpath_mixin = False
+        else:
+            if not cls.lightpath_cpts:
+                raise NotImplementedError(
+                    'Did not implement LightpathMixin properly.  '
+                    'Must supply a list of components (lightpath_cpts)'
+                )
+
+    def get_lightpath_status(self) -> LightpathState:
+        """
+        Create and return a LightpathState object containing information needed
+        for lightpath
+
+        Device logic goes here
+        """
+        raise NotImplementedError('Did not implement LightpathMixin')
+
```

Comment:
> Is that a correct understanding?

Yep, that's exactly what I had in mind.

> splitting get_lightpath_status -> (get_lightpath_status, lightpath_calc)

I like this suggestion, it would make testing easier while also limiting the number of methods users would have to define.  

> making `lp_summary` more useful

I'm wary of bundling too much of the status calculation in the signal itself, since its purpose is more to notify consumers that the signals have changed.  While in the lightpath case, we might want to run the status calculations with every update, I could imagine a case where signals update more frequently than we want to update the gui.  (though I can't imagine the calculations should be that intensive I guess)

I might also be concerned about a single signal containing semi-structured data that needs device knowledge to interpret.  What order do the signal values appear in, do we try to keep that consistent across devices, do users start to use that in other processes, etc.